axon-server-connector/src/main/java/org/axonframework/axonserver/connector/event/LegacyAxonServerEventStorageEngine.java

```diff
@@ -138,10 +133,12 @@ public CompletableFuture<AppendTransaction> appendEvents(@Nonnull AppendConditio
                                                                          .build())
                                              .setMessageIdentifier(event.getIdentifier())
                                              .setTimestamp(event.getTimestamp().toEpochMilli());
+                String aggregateIdentifier = resolveAggregateIdentifier(taggedEvent.tags());
+                String aggregateType = resolveAggregateType(taggedEvent.tags());
```

Comment:
I do think we would benefit if the potential exceptions thrown by the `resolveAggregateIdentifier` and `resolveAggregateType` could reference the event in question that caused the failure for the entire `events` `List` that's given. At least, I think that'll help us and users in the future when they hit this predicament.

Reply:
That's indeed nice information. I'll add it to the assertValidTags method. That assertion is done before creating the transaction, meaning that effectively, the resolve... methods will never receive an event with more than one tag.